### PR TITLE
Fix cloning with compiled sequential model

### LINKED ISSUE #20884: SKLearnClassifier not working properly
Some day ago I was trying to add a Keras model to scikit-learn Pipeline using SKLearnClassifier (wrapper), but I got this error when doing **.fit()** (notwithstanding the model was already compiled):

_**RuntimeError**: Given model needs to be compiled, and have a loss and an optimizer_.

So I made this example code to check if the **clone_model** function that the API uses does indeed keep the compiled model:

```python
from keras.layers import Dense, Input
from keras.models import Sequential, clone_model
clf = Sequential()
clf.add(Input((7,)))
clf.add(Dense(8, activation="relu"))
clf.add(Dense(1, activation="sigmoid"))
clf.compile(optimizer="adam", loss="binary_crossentropy", metrics=["accuracy"])
print("Original compiled?", clf.compiled)
cloned = clone_model(clf)
print("Cloned compiled?", cloned.compiled)
```

And the output:
```cmd
Original compiled? True
Cloned compiled? False
```

I think that's probably the reason why SKLearnClassifier is not working well.
Unfortunately, I couldn't find a way to solve this issue.

### PR DESCRIPTION
**Pull request by Surya2k1**
Currently cloning of model is not considering compiled config.

As reported in #20884, the minimal code snippet is as follows.

```
from keras.layers import Dense, Input
from keras.models import Sequential, clone_model
clf = Sequential()
clf.add(Input((7,)))
clf.add(Dense(8, activation="relu"))
clf.add(Dense(1, activation="sigmoid"))
clf.compile(optimizer="adam", loss="binary_crossentropy", metrics=["accuracy"])
print("Original compiled?", clf.compiled)
cloned = clone_model(clf)
print("Cloned compiled?", cloned.compiled)
```

Added a fix for Sequential model and Functional model also.

Fixes #20884 .

Similar issue #20876

### CODE REVIEW

File: keras/src/models/cloning.py [hunk 1]
```diff
@@ -323,6 +323,14 @@ def _clone_sequential_model(model, clone_function, input_tensors=None):
...
             new_layers = [inputs] + new_layers
+    # If model compiled already then set same to cloned model
+    if model.compiled:
+        compiled_config = model.get_compile_config()
+        cloned_model = Sequential(
+            new_layers, name=model.name, trainable=model.trainable
+        )
+        cloned_model.compile_from_config(compiled_config)
+        return cloned_model
     return Sequential(new_layers, name=model.name, trainable=model.trainable)
```
Comment (fchollet) [inline keras/src/models/cloning.py:334]: Please refactor this a bit so that we only have one instantiation of `Sequential`, current code is a bit redundant

**Review by fchollet — CHANGES_REQUESTED** [review]
Thanks for the PR! Looking good!

File: keras/src/models/cloning.py (hunk 1 above)
Comment (Surya2k1) [reply keras/src/models/cloning.py]: Sure.

File: keras/src/models/cloning.py [hunk 2]
```diff
@@ -323,7 +323,16 @@ def _clone_sequential_model(model, clone_function, input_tensors=None):
...
-    return Sequential(new_layers, name=model.name, trainable=model.trainable)
+    cloned_model = Sequential(
+        new_layers, name=model.name, trainable=model.trainable
+    )
+
+    # If model compiled already then set same to cloned model
+    if model.compiled:
+        compiled_config = model.get_compile_config()
+        cloned_model.compile_from_config(compiled_config)
+        return cloned_model
```
Comment (fchollet) [inline keras/src/models/cloning.py:334]: This line can be removed

Comment (Surya2k1) [reply keras/src/models/cloning.py:334]: My bad. Removed the unwanted line.

**Review by fchollet — APPROVED** [review]
